internal/drivers/ankabuild/driver.go

```diff
@@ -138,6 +107,73 @@ func (c *config) Create(ctx context.Context, opts *types.InstanceCreateOpts) (in
 	return instance, nil
 }
 
+func (c *config) CreateVM(ctx context.Context, request *createVMParams, maxRetries int, retryInterval time.Duration) (*vmResponse, error) {
+	var vm *vmResponse
+	retry := 0
+	for retry < maxRetries {
+		response, err := c.ankaClient.VMCreate(ctx, request)
+		if err != nil {
+			if retry < maxRetries {
+				logrus.Infof("ankabuild: failed to create vm, retrying in %v seconds", retryInterval.Seconds())
+				retry++
+				time.Sleep(retryInterval)
+				retryInterval = retryInterval * 2
+				continue
+			} else {
+				return nil, fmt.Errorf("failed to create vm after %d retries: %v", maxRetries, err)
+			}
+		}
+		var id = response.Body[0]
+		vm = &vmResponse{}
+		vm, _ = c.FindVM(ctx, id, retryInterval)
```

Comment:
This logic is not fully clear to me. So `CreateVM` is taking a `retryInterval` which seems to be a backoff interval and doubles on each retry - but why is this passed into the `FindVM` function - which is treating the `retryInterval` as max time it will retry?

Reply:
I basically just reused the retry Interval rather than creating 2 distinct ones 'retryIntervalCreate & retryIntervalfind'

Reply:
in the case of find vm it's used for the maximum time that the function will keep trying

Reply:
I've separated these out to stop confusion